Java/core/src/main/java/com/amazon/randomcutforest/store/PointStore.java

```diff
@@ -160,13 +161,11 @@ public int decrementRefCount(int index) {
      */
     int takeIndex() {
         if (indexManager.isFull()) {
-            if (indexManager.getCapacity() < capacity) {
+            if (indexManager.getCapacity() < capacity && dynamicResizingEnabled) {
```

Comment:
I think the error message in the `else` branch is misleading. The problem is that the point store has reached its capacity, right?